pytest_fixtures/core/contenthosts.py

```diff
@@ -350,3 +351,115 @@ def custom_host(request):
     deploy_args['workflow'] = 'deploy-rhel'
     with Broker(**deploy_args, host_class=Satellite) as host:
         yield host
+
+
+@pytest.fixture
+def fake_yum_repos(request, target_sat, module_org, module_ak, module_cv, module_product):
+    """Create and sync 3 YUM repositories, with fake custom content, to target satellite."""
+    repositories = {
+        'yum_0': settings.repos.yum_0.url,
+        'yum_6': settings.repos.yum_6.url,
+        'yum_9': settings.repos.yum_9.url,
+    }
+    for repo, url in repositories.items():
+        r = target_sat.api.Repository(
+            product=module_product,
+            url=url,
+        ).create()
+        r.sync()
+        repositories[repo] = r.read()
+    yield repositories.values()
+    # cleanup: delete the created repositories,
+    # first, remove environment and content-view associations from AK,
+    # then, delete all of the published version(s) which may contain the repos,
+    module_ak = module_ak.read()
+    module_ak.environment = []
+    module_ak.content_view = None
+    module_ak.update(['content_view', 'environment'])
+    module_cv = module_cv.read()
+    task = module_cv.remove_version(versions=module_cv.version)
+    assert target_sat.api.ForemanTask(id=task['id']).poll()
+    # delete the updated repositories
+    for repo in repositories.values():
+        repo.read().delete()
+
+
+@pytest.fixture
+def registered_contenthosts(request, target_sat, module_org, module_ak, module_cv, fake_yum_repos):
+    """Create and register multiple contenthosts to satellite.
+    Parametrize int `count` (num of registered hosts), default: 2.
+        by passing a request containing param 'count'.
+    Parametrize int `rhel` (deploy_rhel_version), default: DEFAULT_RHEL_VERSION.
+        by passing a request containing param 'rhel'.
+    Parametrize str `workflow` (used for checkout), default: 'deploy-template'.
+        by passing a request containing param 'workflow'.
+
+    example: No parametrization will result in 2 hosts of DEFAULT RHEL Version.
+
+    eg. distro=rhel10, count=5, default workflow:
+        @pytest.mark.parametrize(
+            'registered_contenthosts',
+            [{'rhel': 10, 'count': 5}],
+            indirect=True,
+        )
+
+    Note: Does Not make use of `markers.rhel_ver_match`,
+        or `markers.rhel_ver_list`, leads to duplicate parametrizations.
+    Three custom repositories will be synced & enabled initially,
+    Add, enable, and sync any other repos to CV/AK for hosts, following fixture setup.
+    """
+    _config = host_conf(request)
+    param = getattr(request, 'param', {})
+    count = param.get('count', 2)
```

Comment:
If you set this here, you don't have to explicitly pass the count as a separate param to broker later.
```suggestion
    _config["count"] = param.get('count', 2)
```